sljit_src/sljitLir.c

```diff
@@ -1380,6 +1380,82 @@ static SLJIT_INLINE CHECK_RETURN_TYPE check_sljit_emit_op1(struct sljit_compiler
 	CHECK_RETURN_OK;
 }
 
+static SLJIT_INLINE CHECK_RETURN_TYPE check_sljit_emit_atomic_load(struct sljit_compiler *compiler, sljit_s32 op,
+	sljit_s32 base_reg, sljit_s32 data_reg, sljit_s32 temp_reg)
+{
+	if (SLJIT_UNLIKELY(compiler->skip_checks)) {
+		compiler->skip_checks = 0;
+		CHECK_RETURN_OK;
+	}
+
+#if (defined SLJIT_ARGUMENT_CHECKS && SLJIT_ARGUMENT_CHECKS)
+	CHECK_ARGUMENT(SLJIT_MOV <= GET_OPCODE(op) && GET_OPCODE(op) <= SLJIT_MOV32 && ((GET_OPCODE(op) % 2) == 1 || GET_OPCODE(op) == SLJIT_MOV));
```

Comment:
GET_OPCODE(op) >= SLJIT_MOV && GET_OPCODE(op) <= SLJIT_MOV_P
Four of them can have SLJIT_32 flag, the rest cannot.